audio-format-service/tests/test_main.py

```diff
@@ -0,0 +1,295 @@
+        mock_content = b"not audio content"
+        files = {"file": ("test.txt", io.BytesIO(mock_content), "text/plain")}
+
+        response = self.client.post("/validate", files=files)
+
+        # Should handle invalid content gracefully
+        assert response.status_code in [200, 400]
+
+
+class TestAudioFormatServiceIntegration:
+    """Integration tests for service functionality."""
+
+    def setup_method(self):
+        """Set up test fixtures before each test method."""
+        # Import here to ensure proper path setup
+        import main
+        from stt_shared.mock_audio_format_handler import MockAudioFormatHandler
+
+        # Initialize the handler manually for testing
+        main.audio_handler = MockAudioFormatHandler()
+
+        self.client = TestClient(app)
+
+    def test_workflow_validate_then_convert(self):
+        """Test complete workflow: validate then convert."""
+        mock_audio_content = b"mock audio content"
+        files = {"file": ("test_audio.wav", io.BytesIO(mock_audio_content), "audio/wav")}
+
+        # First validate
+        validate_response = self.client.post("/validate", files=files)
+        assert validate_response.status_code == 200
+
+        # Reset file pointer for conversion
+        files = {"file": ("test_audio.wav", io.BytesIO(mock_audio_content), "audio/wav")}
+        data = {"target_format": "mp3"}
+
+        # Then convert
+        convert_response = self.client.post("/convert", files=files, data=data)
+        assert convert_response.status_code == 200
+
+    def test_workflow_detect_then_metadata(self):
+        """Test complete workflow: detect format then get metadata."""
+        mock_audio_content = b"mock audio content"
+        files = {"file": ("test_audio.flac", io.BytesIO(mock_audio_content), "audio/flac")}
+
+        # First detect format
+        detect_response = self.client.post("/detect", files=files)
+        assert detect_response.status_code == 200
+        detected_format = detect_response.json()["format"]
+
+        # Reset file pointer for metadata
+        files = {"file": ("test_audio.flac", io.BytesIO(mock_audio_content), "audio/flac")}
+
+        # Then get metadata
+        metadata_response = self.client.post("/metadata", files=files)
+        assert metadata_response.status_code == 200
+        metadata = metadata_response.json()
+
+        # Format should match
+        assert metadata["format"] == detected_format
+
+    def test_service_startup_and_shutdown(self):
+        """Test service startup and shutdown behavior."""
+        # Test that service is responsive after startup
+        response = self.client.get("/health")
+        assert response.status_code == 200
+
+        # Test that supported formats are available
+        response = self.client.get("/formats")
+        assert response.status_code == 200
+        assert len(response.json()) > 0
+
+
+if __name__ == "__main__":
+    pytest.main([__file__])
```

Comment:
why do we need it? Shouldn't pytest run it without this extra help? If not we might need to fix pytest setup instead of trying do this work around